api/src/main/java/org/pmiops/workbench/cdr/dao/CustomCBCriteriaDaoImpl.java

```diff
@@ -59,21 +61,50 @@ public MapSqlParameterSource getParameters() {
           + "and (%s)\n"
           + "order by est_count desc, name asc\n";
 
+  private static final String AUTO_COMPLETE_ENDS_WITH_WITHOUT_TERM =
+      "select *\n"
+          + "from %s.cb_criteria\n"
+          + "where type = :type\n"
```

Comment:
I would use the constants you created in the sql statements as well.
```suggestion
          + "where type = :" + BIND_VAR_TYPE + "\n"
```

Reply:
done